### PR TITLE
Dynamic port configuration - add port buffer cfg to the port ref counter

### PR DESCRIPTION
**Pull request by tomer-israel**
**What I did**
I added increasing/decreasing to the port ref counter each time a port buffer configuration is added or removed

**Why I did it**
In order to avoid cases where a port is removed before the buffer configuration on this port were removed also

**How I verified it**
I added a vs test that test it.
we remove a port with buffer configuration and the port is not removed. only after all buffer cfg on this port were removed this port is removed.

**Details if related**

### CODE REVIEW

File: orchagent/bufferorch.cpp [hunk 1]
```diff
@@ -808,6 +811,36 @@ task_process_status BufferOrch::processQueue(KeyOpFieldsValuesTuple &tuple)
...
+    /* when we apply buffer configuration we need to increase the ref counter of this port
+     * or decrease the ref counter for this port when we remove buffer cfg
+     * so for each priority cfg in each port we will increase/decrease the ref counter
+     * also we need to know when the set command is for creating a buffer cfg or modifying buffer cfg -
+     * we need to increase ref counter only on create flow.
+     * so we added a map that will help us to know what was the last command for this port and priority -
+     * if the last command was set command then it is a modify command and we dont need to increase the buffer counter
+     * all other cases (no last command exist or del command was the last command) it means that we need to increase the ref counter */
+    for (const auto &port_name : port_names)
+    {
```
[the remark this answers is not in the document]
Comment (tomer-israel) [reply orchagent/bufferorch.cpp]: i changed it according to your suggestion

File: orchagent/bufferorch.cpp [hunk 2]
```diff
@@ -808,6 +811,36 @@ task_process_status BufferOrch::processQueue(KeyOpFieldsValuesTuple &tuple)
...
+        for (unsigned int ind = range_low; ind <= range_high; ind++)
+        {
+            if (op == SET_COMMAND) {
+                if (queue_port_flags[port_name][ind] == SET_COMMAND) {
+                    /* if the last operation was "SET_COMMAND" then it's modify and not create - no need to increase ref counter */
+                    continue;
+                }
+                gPortsOrch->increasePortRefCount(port_name);
+            } else if (op == DEL_COMMAND){
```
[the remark this answers is not in the document]
Comment (tomer-israel) [reply orchagent/bufferorch.cpp]: I added to the del operation a "protection" - a decrease to ref count will be only in the case that a set command operation was the last executed command

**Review by stephenxs — COMMENTED** [review]
The change in buffer orchagent looks good to me.
I see there are other changes in the PR as well, please contact corresponding owners to review them if they were not pushed by mistaken

**Comment by tomer-israel** [comment]
added by mistake more changes to file, i created a new PR which is based on the buffer changes here:
https://github.com/tomer-israel/sonic-swss/pull/10

**Pull request closed without merging**